code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -305,6 +305,24 @@
 	required_temp = 374
 	mob_react = FALSE
 
+/datum/chemical_reaction/ez_smoke
+	name = /datum/reagent/smoke_powder
+	results = list(/datum/reagent/smoke_powder = 1)
+	required_reagents = list(/datum/reagent/smoke_powder = 1, /datum/reagent/smoke_powder_activator = 1)
```

Comment:
/datum/reagent/smoke_powder_activator should be a catalyst so 1 unit activates all the smoke powder

Reply:
This creates an infinite loop error because the output reagent is the same as the input. 

I can just make a subtype to bypass this though I suppose.